apps/linode-marketplace-passbolt/roles/passbolt/tasks/main.yml

```diff
@@ -0,0 +1,219 @@
+
+- name: Run gpg --list-secret-keys and save output to a temp file
+  shell: gpg --list-secret-keys > /tmp/secret-keys.txt
+  args:
+    executable: /bin/bash
+
+- name: Extract the key ID and save to a variable
+  shell: |
+    awk '/^sec/{getline; print $1}' /tmp/secret-keys.txt
+  register: gpg_key_id_output
+
+- name: Set the extracted key ID as a fact
+  set_fact:
+    gpg_secret_key: "{{ gpg_key_id_output.stdout }}"
+
+- name: Display GPG secret key variable
+  debug:
+    msg: "The GPG secret key ID is: {{ gpg_secret_key }}"
+
+- name: Writing gpg_secret_key to group_vars/linode/vars
+  lineinfile:
+    path: group_vars/linode/vars
+    line: "gpg_secret_key: '{{ gpg_secret_key }}'"
+
+- name: Copy passbolt config template to /etc/passbolt/passbolt.php
+  template:
+    src: templates/passbolt.php
+    dest: /etc/passbolt/passbolt.php
+    owner: www-data
+    group: www-data
+    mode: '0440'
+
+- name: Update GPG fingerprint in passbolt.php
+  ansible.builtin.replace:
+    path: /etc/passbolt/passbolt.php
+    regexp: "'fingerprint' => '.*',"
+    replace: "'fingerprint' => '{{ gpg_secret_key }}',"
+
+- name: Export GPG public key to serverkey.asc
+  shell: >
+    gpg --armor --export {{ gpg_secret_key }} > /etc/passbolt/gpg/serverkey.asc
+  args:
+    executable: /bin/bash
+
+- name: Export GPG private key to serverkey_private.asc
+  shell: >
+    gpg --armor --export-secret-keys {{ gpg_secret_key }} > /etc/passbolt/gpg/serverkey_private.asc
+  args:
+    executable: /bin/bash
+
+- name: Install Passbolt without admin user
+  become: yes
+  become_method: sudo
+  become_user: "{{ webuser }}"
+  shell: "/usr/share/php/passbolt/bin/cake passbolt install --no-admin --force"
+  args:
+    executable: /bin/bash
+
+- name: Register first admin user for Passbolt
+  become: yes
+  become_method: sudo
+  become_user: "{{ webuser }}"
+  shell: "/usr/share/php/passbolt/bin/cake passbolt register_user -u {{ passbolt_first_admin_email }} -f {{ passbolt_first_admin_username }} -l {{ passbolt_first_admin_surname }} -r admin"
+  args:
+    executable: /bin/bash
+  register: admin_registration_output
+
+- name: Extract registration URL
+  set_fact:
+    passbolt_admin_registration_url: "{{ admin_registration_output.stdout | regex_search('https://' + _domain | regex_escape + '/setup/start/[a-z0-9-]+/[a-z0-9-]+') }}"
+
+- name: Writing passbolt_admin_registration_url to group_vars/linode/vars
+  lineinfile:
+    path: group_vars/linode/vars
+    line: "passbolt_admin_registration_url: '{{ passbolt_admin_registration_url }}'"
+
+- name: Set ownership and permissions for serverkey.asc
+  ansible.builtin.file:
+    path: /etc/passbolt/gpg/serverkey.asc
+    owner: www-data
+    group: www-data
+    mode: '0440'  
+    state: file
+
+- name: Set ownership and permissions for serverkey_private.asc
+  ansible.builtin.file:
+    path: /etc/passbolt/gpg/serverkey_private.asc
+    owner: www-data
+    group: www-data
+    mode: '0440'  
+    state: file
```

Comment:
we can consolidate this into a single task by using `loop` :)

Reply:
Done.